### PR TITLE
[Hub Generated] Review request for Microsoft.Network to add version stable/2020-11-01

### PR DESCRIPTION
**Pull request by disreedh**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/disreedh/azure-rest-api-specs/disreedh-dev-frontdoor-Microsoft.Network-2020-11-01...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/frontdoor.json [hunk 1]
```diff
@@ -1,2350 +0,0 @@
...
-      "type": "string",
-      "pattern": "^[a-zA-Z0-9]+(-*[a-zA-Z0-9])*$",
-      "x-previous-pattern": "^[-\\w\\d\\._]+$",
-      "minLength": 1,
-      "maxLength": 90,
-      "x-ms-parameter-location": "method",
-      "description": "Name of the Rules Engine which is unique within the Front Door."
-    }
-  }
-}
```
Comment (akning-ms) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/frontdoor.json]: Why front door removed from this version?

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json [hunk 2]
```diff
@@ -270,8 +270,15 @@
...
           "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        },
+        "sku": {
+          "description": "The pricing tier of web application firewall policy.",
+          "$ref": "#/definitions/Sku"
         }
       },
+      "required": [
+        "sku"
+      ],
```
Comment (akning-ms) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json:281]: Add a required property is a breaking change, even in a new version. according to breaking change policy: http://aka.ms/bcforapi
can it be optional?

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json [hunk 3]
```diff
@@ -711,6 +749,38 @@
           "readOnly": true,
           "description": "Version of the managed rule set type."
         },
+        "ruleSetVersionState": {
```
Comment (eschwabe) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json:752]: Should be read-only.

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. Also Please go through the ARM API Review Checklist. These steps are required before review can be continued.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/frontdoor.json (hunk 1 above)
Comment (disreedh) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/frontdoor.json]: This new api changes are only related to web application firewall policy. So not including frontdoor.json

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json (hunk 2 above)
Comment (disreedh) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json]: Thank you for pointing that out. I have changed Sku to optional.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json (hunk 3 above)
Comment (disreedh) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json]: This field has been removed from 20201101 version

**Comment by pilor** [comment]
NRP usually uses monthly release branches for swagger updates.  Is there a reason this isn't following the normal NRP process?

**Review by pilor — APPROVED** [review]

**Comment by majastrz** [comment]
Removed the breaking changes review label as breaking change rules passed and I don't see any breaking changes issues. @akning-ms please confirm. The PR is still good from ARM side.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by akning-ms** [comment]
@msyyc  for confirm Pythons SDK breaking change

**Comment by akning-ms** [comment]
Hi @disreedh, Maybe we discussed, but can not recall, Can you comment what is conclusion for below lint failure?:

❌ R3023 - OperationsAPIImplementation | Operations API must be implemented for '/providers/Microsoft.Network/operations'.New: Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json#L37
-- | --

**Comment by msyyc** [comment]
> @msyyc for confirm Pythons SDK breaking change

Breaking change is not caused by this PR.

**Comment by disreedh** [comment]
@akning-ms Sorry I think I did not loop you in offline chats I had with ARM review DRI. For Frontdoor WAF, the operations API is defined under a different path (it's part of the Microsoft.Network APIs) azure-rest-api-specs/operation.json at master · Azure/azure-rest-api-specs (github.com). 

The DRI asked me to suppress the issue for approval. For a permanent suppression I have raised a ticket following https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process for ARM Violation suppression.

**Pull request merged**